src/modules/did/did.types.ts

```diff
@@ -2,24 +2,49 @@
  * A value object representing EIP-1056 DID
  */
 export class DID {
-  constructor(id: string) {
-    const idParts = id.split(':');
+  /**
+   * A DID in the format of "did:" method-name ":" method-specific-id
+   */
+  readonly did: string;
+
+  readonly method: 'ethr' | string;
+
+  readonly id: string;
+
+  readonly chain?: string;
+
+  constructor(did: string) {
+    const idParts = did.split(':');
     if (idParts.length < 3) {
       throw new Error('DID should consists of at least 3 components');
     }
-    this.id = id;
-    const [, method] = idParts;
-    this.method = method;
-  }
 
-  /**
-   * A DID in the format of did:ethr:{ethereumAddress}
-   */
-  readonly id: string;
-  readonly method: 'ethr' | string;
+    let didMethod: string;
+    let didChain: string | undefined;
+    let didId: string;
+
+    // Back compatibility with old format
+    if (idParts.length === 3) {
+      didMethod = idParts[1];
+      didChain = undefined;
+      didId = idParts[2];
+    } else if (idParts.length === 4) {
+      didMethod = idParts[1];
+      didChain = idParts[2];
+      didId = idParts[3];
+    } else {
+      throw new Error('Unsupported DID format');
+    }
```

Comment:
I think this code looks good but wondering if you've seen similar code elsewhere? I wonder if this logic could be encapsulated in ew-did-registry as there is probably similar logic there, I think 🤔

Reply:
I copied it from `did-registry` since there was no export for DID class

Reply:
Ok, I will export now so that you can use, I think that is better than introducing this duplication

Reply:
Mmm, actually, let's do that afterwards. Let's get this merged for now